### PR TITLE
VS 2019 + different c++ standards

### PR DESCRIPTION
**Pull request by kanstantsin-chernik**
Problem:
- VS 2019 is not tested in appveyor
- Build system doesn't support variations of c++ standards

Solution:
- Add VS 2019 into appveyor
- Add C++ standards as a parameter into makefile and CMakeList

Issue: #

Reviewers:
@

### CODE REVIEW

File: test/ft/di_errors.cpp [hunk 1]
```diff
@@ -414,9 +414,9 @@ test bind_any_of_not_related = [] {
 #if defined(__cpp_variable_templates)
 test bind_any_of_not_related_v = [] {
   auto errors_ = errors("constraint not satisfied",
-#if defined(__MSVC__) && _MSC_VER >= 1916
+#if defined(__MSVC__) && _MSC_VER >= 1916 && _MSC_VER < 1920
```
Comment (kanstantsin-chernik) [inline test/ft/di_errors.cpp:417]: @krzysztof-jusiak we need to get out of this business. It is just wast of time to fix these ifdefs.

I thing we instead we can just have error as `vector<vector<string>>` and if any of rows is conformed we are good

File: test/common/utils.hpp [hunk 2]
```diff
@@ -49,16 +50,40 @@ std::string cxxflags(bool internal = false) {
...
-    cppflags += "-std=c++1y";
+    compiler = "clang";
 #elif defined(__GCC__)
-    cppflags += "-std=c++1y";
-#elif defined(__MSVC__) && _MSVC_LANG >= 201703L
-    cppflags += "/std:c++17";
+    compiler = "gcc";
+#elif defined(__MSVC__)
+    compiler = "msvc";
 #elif defined(__MSVC__)
```
Comment (kanstantsin-chernik) [inline test/common/utils.hpp]: Will have to fix it tomorrow

File: .appveyor.yml [hunk 3]
```diff
@@ -28,26 +27,94 @@ environment:
...
+  # - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
+  #   VSPATH: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build
+  #   BS: cmake
+  #   CMAKE_GENERATOR: Visual Studio 16 2019
+  #   MEMCHECK: DRMEMORY
+  #   PLATFORM: amd64
+  #   CXX_STANDARD: 20
+
+  # - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019 Preview
+  #   VSPATH: C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Auxiliary\Build
```
Comment (kanstantsin-chernik) [inline .appveyor.yml:89]: Preview image cmake does not support vs 2019

File: .appveyor.yml [hunk 4]
```diff
@@ -28,26 +27,94 @@ environment:
...
+    PLATFORM: amd64
+    CXX_STANDARD: 17
+
+  # - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
+  #   VSPATH: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build
+  #   BS: cmake
+  #   CMAKE_GENERATOR: Visual Studio 16 2019
+  #   MEMCHECK: DRMEMORY
+  #   PLATFORM: amd64
+  #   CXX_STANDARD: 20
```
Comment (kanstantsin-chernik) [inline .appveyor.yml:86]: We still have a comile error here

File: .appveyor.yml [hunk 5]
```diff
@@ -28,26 +27,94 @@ environment:
...
+    PLATFORM: amd64
+    CXX_STANDARD: 14
 
   - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017 Preview
     VSPATH: C:\Program Files (x86)\Microsoft Visual Studio\Preview\Community\VC\Auxiliary\Build
     BS: cmake
     CMAKE_GENERATOR: Visual Studio 15 2017
     MEMCHECK: DRMEMORY
     PLATFORM: amd64
+    CXX_STANDARD: 17
```
Comment (kanstantsin-chernik) [inline .appveyor.yml:62]: Not sure about c++20 for vs 2017

**Comment by kris-jusiak** [comment]
Fantastic, great job @kanstantsin-chernik :+1:

**Pull request merged**